catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/BackupCommandTest.java

```diff
@@ -0,0 +1,976 @@
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        setDdfHome();
+        createDefaultMiniSolrCloudCluster();
+        addDocument("1");
+    }
+
+    @Before
+    public void before() throws Exception {
+        cipherSuites = System.getProperty("https.cipherSuites");
+        System.setProperty("https.cipherSuites",
+                "TLS_DHE_RSA_WITH_AES_128_CBC_SHA,TLS_DHE_RSA_WITH_AES_128_CBC_SHA,TLS_DHE_DSS_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA");
+        protocols = System.getProperty("https.protocols");
+        System.setProperty("https.protocols", "TLSv1.1, TLSv1.2");
+        consoleOutput = new ConsoleOutput();
+        consoleOutput.interceptSystemOut();
+
+        mockHttpWrapper = mock(HttpWrapper.class);
+    }
+
+    @After
+    public void after() {
+        consoleOutput.resetSystemOut();
+
+        System.clearProperty(SOLR_CLIENT_PROP);
+        System.clearProperty(ZOOKEEPER_HOSTS_PROP);
+
+        if (cipherSuites != null) {
+            System.setProperty("https.cipherSuites", cipherSuites);
+        } else {
+            System.clearProperty("https.cipherSuites");
+        }
+        if (protocols != null) {
+            System.setProperty("https.protocols", protocols);
+        } else {
+            System.clearProperty("https.protocols");
+        }
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        if (miniSolrCloud != null) {
+            miniSolrCloud.getSolrClient()
+                    .close();
+            miniSolrCloud.shutdown();
+        }
+    }
+
+    @Test
+    public void testNoArgBackup() throws Exception {
+
+        when(mockHttpWrapper.execute(any(URI.class))).thenReturn(mockResponse(HttpStatus.SC_OK,
+                ""));
+
+        BackupCommand backupCommand = new BackupCommand() {
+            @Override
+            protected HttpWrapper getHttpClient() {
+                return mockHttpWrapper;
+            }
+        };
+        backupCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format("Backup of [%s] complete.", DEFAULT_CORE_NAME)));
+
+    }
+
+    @Test
+    public void testBackupSpecificCore() throws Exception {
+        final String coreName = "core";
+
+        when(mockHttpWrapper.execute(any(URI.class))).thenReturn(mockResponse(HttpStatus.SC_OK,
+                ""));
+
+        BackupCommand backupCommand = new BackupCommand() {
+            @Override
+            protected HttpWrapper getHttpClient() {
+                return mockHttpWrapper;
+            }
+        };
+
+        backupCommand.coreName = coreName;
+        backupCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format("Backup of [%s] complete.", coreName)));
+
```

Comment:
Leaving as-is since this is an existing test and was not added/modified for this ticket.